Relative paths aren't evaluated properly in symlinked files

I'm trying to include the `devcontainer` command in my Docker image. I extend a Next.js a standalone build by setting `outputFileTracingIncludes` to the results of `nodeFileTrace()` for the additional binaries I want to bundle.

Anyway, the issue is easily reproducable by installing `@devcontainers/cli` **locally** and then:
```
$ node_modules/.bin/nft print node_modules/.bin/devcontainer 
FILELIST:
node_modules/.bin/devcontainer
node_modules/@devcontainers/cli/devcontainer.js


WARNINGS:
Error: Failed to resolve dependency "./dist/spec-node/devContainersSpecCLI":
Cannot find module '/home/dev/vibe-web/node_modules/.bin/dist/spec-node/devContainersSpecCLI' loaded from /home/dev/vibe-web/node_modules/.bin/devcontainer
```

As usual, `node_modules/.bin/devcontainer` is just a symlink to `../@devcontainers/cli/devcontainer.js`, which is really just:
```js
#!/usr/bin/env node
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

require('./dist/spec-node/devContainersSpecCLI');
```

Seemingly, this relative path should be evaluated based on the actual file location / symlink target (`../@devcontainers/cli/`). Here's the test that it would work (despite some strange warnings that I assume to be caused by some settings in the `nft` CLI):
```
$ node_modules/.bin/nft print node_modules/.bin/../@devcontainers/cli/devcontainer.js
Skipping asset emission of /home/dev/vibe-web/*/* for /home/dev/vibe-web/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js as it is outside the package base /home/dev/vibe-web/node_modules/@devcontainers/cli
Skipping asset emission of /home/dev/vibe-web/*/* for /home/dev/vibe-web/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js as it is outside the package base /home/dev/vibe-web/node_modules/@devcontainers/cli
Skipping asset emission of /home/dev/vibe-web/*/* for /home/dev/vibe-web/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js as it is outside the package base /home/dev/vibe-web/node_modules/@devcontainers/cli
Skipping asset emission of /tmp for /home/dev/vibe-web/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js as it is outside the package base /home/dev/vibe-web/node_modules/@devcontainers/cli
FILELIST:
node_modules/@devcontainers/cli/devcontainer.js
node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js
node_modules/@devcontainers/cli/package.json
```

--- a/src/node-file-trace.ts
+++ b/src/node-file-trace.ts
@@ -329,63 +329,69 @@ export class Job {
   ) {
     if (depth < 0)
       throw new Error('invariant - depth option cannot be negative');
-    if (this.processed.has(path)) {
+
+    // Resolve symlinks so that dependencies are resolved relative to the real
+    // file location, not the symlink location
+    const realPath = await this.realpath(path, parent);
+
+    if (this.processed.has(realPath)) {
       if (parent) {
         await this.emitFile(path, 'dependency', parent);
       }
       return;
     }
-    this.processed.add(path);
+    this.processed.add(realPath);
 
     // Additional dependencies.
-    const additionalDeps = this.remappings.get(path);
+    const additionalDeps = this.remappings.get(realPath);
     if (additionalDeps) {
       await Promise.all(
         [...additionalDeps].map(async (dep) =>
-          this.emitDependency(dep, path, depth),
+          this.emitDependency(dep, realPath, depth),
         ),
       );
     }
 
     const emitted = await this.emitFile(path, 'dependency', parent);
     if (!emitted) return;
-    if (path.endsWith('.json')) return;
-    if (path.endsWith('.node')) return await sharedLibEmit(path, this);
+    if (realPath.endsWith('.json')) return;
+    if (realPath.endsWith('.node')) return await sharedLibEmit(realPath, this);
 
     // .js and .ts files can change behavior based on { "type": "module" }
     // in the nearest package.json so we must emit it too. We don't need to
     // emit for .cjs/.mjs/.cts/.mts files since their behavior does not
     // depend on package.json
-    if (path.endsWith('.js') || path.endsWith('.ts')) {
-      const pjsonBoundary = await this.getPjsonBoundary(path);
+    if (realPath.endsWith('.js') || realPath.endsWith('.ts')) {
+      const pjsonBoundary = await this.getPjsonBoundary(realPath);
       if (pjsonBoundary)
         await this.emitFile(
           pjsonBoundary + sep + 'package.json',
           'resolve',
-          path,
+          realPath,
         );
     }
     if (depth === 0) return;
 
     let analyzeResult: AnalyzeResult;
 
-    const cachedAnalysis = this.analysisCache.get(path);
+    const cachedAnalysis = this.analysisCache.get(realPath);
     if (cachedAnalysis) {
       analyzeResult = cachedAnalysis;
     } else {
-      const source = await this.readFile(path);
-      if (source === null) throw new Error('File ' + path + ' does not exist.');
+      const source = await this.readFile(realPath);
+      if (source === null)
+        throw new Error('File ' + realPath + ' does not exist.');
       // analyze should not have any side-effects e.g. calling `job.emitFile`
       // directly as this will not be included in the cachedAnalysis and won't
       // be emit for successive runs that leverage the cache
-      analyzeResult = await analyze(path, source.toString(), this);
-      this.analysisCache.set(path, analyzeResult);
+      analyzeResult = await analyze(realPath, source.toString(), this);
+      this.analysisCache.set(realPath, analyzeResult);
     }
 
     const { deps, imports, assets, isESM } = analyzeResult;
 
     if (isESM) {
-      this.esmFileList.add(relative(this.base, path));
+      this.esmFileList.add(relative(this.base, realPath));
     }
 
     await Promise.all([
@@ -403,14 +409,14 @@ export class Job {
               .slice(this.base.length)
               .indexOf(sep + 'node_modules' + sep) === -1)
         )
-          await this.emitDependency(asset, path, depth - 1);
-        else await this.emitFile(asset, 'asset', path);
+          await this.emitDependency(asset, realPath, depth - 1);
+        else await this.emitFile(asset, 'asset', realPath);
       }),
       ...[...deps].map(async (dep) =>
-        this.maybeEmitDep(dep, path, !isESM, depth - 1),
+        this.maybeEmitDep(dep, realPath, !isESM, depth - 1),
       ),
       ...[...imports].map(async (dep) =>
-        this.maybeEmitDep(dep, path, false, depth - 1),
+        this.maybeEmitDep(dep, realPath, false, depth - 1),
       ),
     ]);
   }

--- a/test/unit/require-symlink-subdir/input.js
+++ b/test/unit/require-symlink-subdir/input.js
@@ -1,0 +1,1 @@
+subdir/input.js

--- a/test/unit/require-symlink-subdir/output.js
+++ b/test/unit/require-symlink-subdir/output.js
@@ -1,0 +1,6 @@
+[
+  "package.json",
+  "test/unit/require-symlink-subdir/input.js",
+  "test/unit/require-symlink-subdir/subdir/another.js",
+  "test/unit/require-symlink-subdir/subdir/input.js"
+]

--- a/test/unit/require-symlink-subdir/subdir/another.js
+++ b/test/unit/require-symlink-subdir/subdir/another.js
@@ -1,0 +1,1 @@
+module.exports = () => console.log('hello');

--- a/test/unit/require-symlink-subdir/subdir/input.js
+++ b/test/unit/require-symlink-subdir/subdir/input.js
@@ -1,0 +1,1 @@
+require('./another.js')

